### PR TITLE
core: fix 'pxj' command to not use core->block

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

`pxj` was using `core->block`, while `px` is correctly reading the length as requested. Fixing it by using the same code as `px`.

**Test plan**

```
$ rizin /bin/ls
> px 272
> pxj 272
```
output should be the same (no 0 values at the end of the `pxj` output).

**Closing issues**

Address the problem in https://github.com/rizinorg/rizin/issues/3650#issuecomment-1652237530

### CODE REVIEW

File: librz/core/cprint.c [hunk 1]
```diff
@@ -346,32 +346,39 @@ RZ_API RZ_OWN char *rz_core_print_hexdump_or_hexdiff_str(RZ_NONNULL RzCore *core
...
-			rz_io_read_at(core->io, addr, buffer, len);
+	ut64 from = rz_config_get_i(core->config, "diff.from");
+	ut64 to = rz_config_get_i(core->config, "diff.to");
+	if (from == to && !from) {
+		len_fixup(core, &addr, &len);
+		ut8 *buffer = malloc(len);
+		if (!buffer) {
+			return NULL;
+		}
+		rz_io_read_at(core->io, addr, buffer, len);
```
Comment (XVilka) [inline librz/core/cprint.c:357]: It can fail, please handle that

**Review by XVilka — CHANGES_REQUESTED** [review]

**Comment by wargio** [comment]
I think pxj was correctly behaving, and px is incorrect. we should not ask the user to provide the length but we should ask the user to use the `@!<size>` modifier.

**Comment by ret2libc** [comment]
> I think pxj was correctly behaving, and px is incorrect. we should not ask the user to provide the length but we should ask the user to use the `@!<size>` modifier.

Good point... I wonder what's our take on core->block though.

**Comment by wargio** [comment]
i think that all the commands should work in the same way. if i start adding length etc.. it might get messy

**Comment by XVilka** [comment]
Agree on that, let's try to be uniform, especially inside one command "family".

**Comment by ret2libc** [comment]
But length is already present in *all* commands under `px`:
```
Usage: px[?]   # Show hexdump
| px[j] [<len>]  # show hexdump
| pxa [<len>]    # show annotated hexdump
| pxA[l] [<len>] # show op analysis color map
| pxb [<len>]    # dump bits in hexdump form
| pxc [<len>]    # show hexdump with comments
| pxd[jhwq]      # show signed integer dump
| pxe [<len>]    # emoji hexdump! :)
| pxf [<len>]    # show hexdump of current function
| pxi [<len>]    # HexII compact binary representation
| pxr[?]         # show hexword references
| pxs [<len>]    # show hexadecimal in sparse mode
| pxt[j]         # show delta pointer table in rizin commands
| pxx            # show <N> bytes of hex-less hexdump
| pxX [<N>]      # show <N> words of hex-less hexdump
| px0 [<N>]      # 8bit hexpair list of bytes until zero byte
| pxh[j] [<len>] # show 2-bytes hexadecimal integers dump
| pxH[j] [<len>] # show 2-bytes hexadecimal integers dump, one per line
| pxw[j] [<len>] # show 4-bytes hexadecimal integers dump
| pxW[j] [<len>] # show 4-bytes hexadecimal integers dump, one per line
| pxq[j] [<len>] # show 8-bytes hexadecimal integers dump
| pxQ[j] [<len>] # show 8-bytes hexadecimal integers dump, one per line
| pxo [<len>]    # show 1-byte octal integers dump
| pxl[j] [<N>]   # display <N> lines of hexdump
```

So in this case it was `pxj` that was not respecting the length. I know however we have used the blocksize in a few places, but I do also think in some others it make sense to keep the length. Like, `px 10` is easier to write and use than `px @!10`.

**Comment by XVilka** [comment]
Lets go with length then

**Comment by wargio** [comment]
agreed.

**Review by wargio — APPROVED** [review]

**Pull request merged**